Core/Core/Assignments/AssignmentDetails/AssignmentDetailsView.swift

```diff
@@ -30,6 +30,8 @@ public struct AssignmentDetailsView: View, ScreenViewTrackable {
     @Environment(\.viewController) var controller
 
     public let screenViewTrackingParameters: ScreenViewTrackingParameters
+    private var isTeacherEnrollment: Bool = false
+    private var isLocked: Bool = false
```

Comment:
I'm not sure if storing these variables here in this form is safe. Both of these come from Store subscriptions which might or might not contain any courses or assignments at the time of the view's init. These variables are populated at this point so the values of these variables could be wrong. Stores are updated in the `onAppear` block so updating these variables from the store update's completion block seems to be a good place after we've changed these variables to `@State`s to actually trigger a UI refresh when they update.

Reply:
Moved them to the completion block.